Core/QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryController.java

```diff
@@ -153,12 +153,74 @@ private void setup() {
 			@RequestParam(name = "options", required = false) List<String> options,
 			@RequestParam(name = "services", required = false) Boolean showServices) {
 		return getQueryData(request, request.getQueryString(), request.getParameterMap(), attrs, limit, offset, qToken,
-				options, showServices, false);
+				options, showServices, false,null);
+	}
+	
+	@GetMapping(path = "/type")
+	public ResponseEntity<byte[]> getAllTypes(HttpServletRequest request, 
+			@RequestParam(value = "details", required = false, defaultValue = "false") boolean details) {
+		String check="NonDeatilsType";
+		if(details==true){
+			check="deatilsType";
+		}
+		ResponseEntity<byte[]> result = getQueryData(request, null, request.getParameterMap(), null, null, null, null, null,
+				false, true,check);
+		if (Arrays.equals(emptyResult1, result.getBody()) || Arrays.equals(emptyResult2, result.getBody())) {
+			return ResponseEntity.status(HttpStatus.NOT_FOUND)
+					.body(new RestResponse(ErrorType.NotFound, "Resource not found.").toJsonBytes());
+		}
+		return result;
+	}
+	
+	@GetMapping(path = "/type/{type}")
```

Comment:
[Minor] It should be like that: /types/{entityType}

Reply:
Done